cpp-package/example/inference/simple_rnn.cpp

```diff
@@ -0,0 +1,359 @@
+  arrayout = arrayout.Reshape(Shape(sequence_length, -1));
+  arrayout = arrayout.ArgmaxChannel();
+  NDArray::WaitAll();
+
+  std::ostringstream oss;
+  for (std::size_t i = 0; i < sequence_length; ++i) {
+    auto charIndex = arrayout.At(0, i);
+    oss << intToWord[charIndex] << " ";
+  }
+  LG << "Output String Predicted by Model: [" << oss.str() << "]";
+}
+
+
+Predictor::~Predictor() {
+  if (executor) {
+    delete executor;
+  }
+  MXNotifyShutdown();
+}
+
+void printUsage() {
+    std::cout << "Usage:" << std::endl;
+    std::cout << "simple_rnn " << std::endl
+              << "[--input] Input string sequence."  << std::endl
+              << "[--gpu]  Specify this option if workflow needs to be run in gpu context "
+              << std::endl;
+}
+
+
+void Download_files(const std::vector<std::string> model_files) {
+  std::string wget_command = "wget -nc ";
+  std::string s3_url = "https://s3.amazonaws.com/mxnet-cpp/RNN_model/";
+  for (auto &file : model_files) {
+    std::ostringstream oss;
+    oss << wget_command << s3_url << file << " -O " << file;
+    system(oss.str().c_str());
+  }
+  return;
+}
+
+int main(int argc, char** argv) {
+  std::string model_file_json = "./obama-speaks-symbol.json";
+  std::string model_file_params ="./obama-speaks-0100.params";
+  std::string input_dictionary = "./obama.dictionary.txt";
+  std::string input_sequence = "But what they did not understand however was that I had to take "
+                "Mr  Keyes seriously for he claimed to speak for my religion and my God";
+  int input_sequence_length = 35;
+  bool gpu_context_type = false;
+
+  int index = 1;
+    while (index < argc) {
+      if (strcmp("--input", argv[index]) == 0) {
+            index++;
+            input_sequence = (index < argc ? argv[index]:input_sequence);
+        } else if (strcmp("--gpu", argv[index]) == 0) {
+            gpu_context_type = true;
+        } else if (strcmp("--help", argv[index]) == 0) {
+            printUsage();
+            return 0;
+        }
+      index++;
+    }
+
+  /*
+   * Download the trained RNN model file, param file and dictionary file.
+   * The dictionary file contains word to index mapping.
+   * Each line of the dictionary file contains a word and an unique index for that word separated
+   * by a space. For example:
+   * snippets 11172
+   * This dictionary file is created when the RNN model was trained with a particular dataset.
+   * Hence the dictionary file is specific to the dataset with which model was trained.
+   */
+  std::vector<std::string> files;
+  files.push_back(model_file_json);
+  files.push_back(model_file_params);
+  files.push_back(input_dictionary);
+
+  Download_files(files);
+
+  try {
+    // Initialize the predictor object
+    Predictor predict(model_file_json, model_file_params, input_dictionary, gpu_context_type,
+                      input_sequence_length);
+
+    // Run the forward pass to predict the image.
+    predict.PredictText(input_sequence);
```

Comment:
Is it not cleaner when Predictor returns the prediction result, and the calling function prints it?
If you were to use a model in production, would you not design it that way?